src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -150,6 +148,11 @@ public abstract class CommandLineArguments
         /// </summary>
         public ImmutableArray<CommandLineAnalyzerReference> AnalyzerReferences { get; internal set; }
 
+        /// <summary>
+        /// A set of paths to EditorConfig-compatible analyzer config files.
+        /// </summary>
+        public ImmutableArray<string> AnalyzerConfigPaths { get; internal set; }
+
```

Comment:
This is confusing. Why do we call them "paths" instead of a filenames? Why do we use `string` instead of `CommandLineSourceFile` like we do for `AdditionalFiles` and `EmbeddedFiles`?

Reply:
Perhaps this is a set of paths to directories containing editor config files?

---
In reply to: [193898640](https://github.com/dotnet/roslyn/pull/27232#discussion_r193898640) [](ancestors = 193898640)

Reply:
A file name usually just refers to the name of the file, i.e. for a file `/a/b/c.cs` the file name would be `c.cs`, while the path would be `/a/b/c.cs`. Right now these happen to be absolute paths, but I didn't want to guarantee that in the API.

As to why I didn't choose to use `CommandLineSourceFile` -- I'm not sure why we *did* choose that for `AdditionalFiles` and `EmbeddedFiles`. `CommandLineSourceFile` is a struct with two entries: `Path`, which is the same as the string here, and `IsScript` which is a bool that is always false, since by definition only source files can be script files. I have a feeling that the helper for parsing separated paths happened to return `CommandLineSourceFiles` and these additions couldn't be bothered to convert. Otherwise I see no reason for them just to be paths.